packages/@css-blocks/core/src/BlockParser/BlockFactory.ts

```
@@ -137,28 +162,47 @@ export class BlockFactory {
     return this._getBlockPromise(identifier);
   }
 
+  /**
+   * Make a promise to load and parse a CSS Block data file, add it to the cache,
+   * and return it.
+   *
+   * @param identifier - An identifier that points at a data file or blob in persistent storage.
+   * @returns A promise that resolves to the parsed block.
+   */
   private _getBlockPromise(identifier: FileIdentifier): Promise<Block> {
     return this.promises[identifier] = this._getBlockPromiseAsync(identifier);
   }
 
+  /**
+   * An async method that loads and parses a CSS Block data file. We load the data here, using
+   * the Importer, then defer to another method to actually parse the data file into a Block.
+   *
+   * @param identifier - An identifier that points at a data file or blob in persistent storage.
+   * @returns A promise that resolves to the parsed block.
+   */
   private async _getBlockPromiseAsync(identifier: FileIdentifier): Promise<Block> {
     try {
       let file = await this.importer.import(identifier, this.configuration);
-      let block = await this._importAndPreprocessBlock(file);
-      debug(`Finalizing Block object for "${block.identifier}"`);
+      if (file.type === "ImportedCompiledCssFile") {
+        // TODO: Process ImportedCompiledCssFile type.
+        return new Block("foo", "bar");
```

This should throw an error until implemented.